### PR TITLE
Fixed a serialization issue, preventing quality level parameters to undo/redo and update scene view on change.

### PR DESCRIPTION
**Pull request by JulienIgnace-Unity**
### Purpose of this PR
Fixed a serialization issue, preventing quality level parameters to undo/redo and update scene view on change.

Fixes: https://fogbugz.unity3d.com/f/cases/1200299/

---
### Testing status
Tested manually the change in the HDRP template scene with bloom settings.

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Editor/RenderPipeline/ScalableSettingLevelParameterEditor.cs [hunk 1]
```diff
@@ -22,13 +22,14 @@ public override bool OnGUI(SerializedDataParameter parameter, GUIContent title)
...
 
-            o.levelAndOverride = SerializedScalableSettingValueUI.LevelFieldGUI(
+            var levelAndOverride = SerializedScalableSettingValueUI.LevelFieldGUI(
                 rect,
                 title,
                 ScalableSettingSchema.GetSchemaOrNull(ScalableSettingSchemaId.With3Levels),
                 level,
                 useOverride
             );
+            value.intValue = levelAndOverride.level;
```
Comment (fredericv-unity3d) [inline com.unity.render-pipelines.high-definition/Editor/RenderPipeline/ScalableSettingLevelParameterEditor.cs:32]: I don't think this works.

If the user overrides the value the level returned is `-1` and if you store this value the behaviour will be undefined.

While reading the volume serialization framework, I see that the `GetObjectRef` return a C# reference of an object, and I don't know if any modification of this object will be applied to the `SerializedProperty`.

I suggest to check how to apply the modifications done to the SerializedDataParameter` to the underlying `SerializedProperty`

Comment (JulienIgnace-Unity) [reply com.unity.render-pipelines.high-definition/Editor/RenderPipeline/ScalableSettingLevelParameterEditor.cs:32]: > While reading the volume serialization framework, I see that the GetObjectRef return a C# reference of an object, and I don't know if any modification of this object will be applied to the SerializedProperty.
It doesn't. That's why we had issues with undo/redo and scene refresh.
> f the user overrides the value the level returned is -1 and if you store this value the behaviour will be undefined
I'm not sure I'm following you. I tried override and it worked just the same as far as I can tell.
From what I see, all other parameters work the same way. See FloatParameterDrawer for example.

Comment (fredericv-unity3d) [reply com.unity.render-pipelines.high-definition/Editor/RenderPipeline/ScalableSettingLevelParameterEditor.cs:32]: Assigning only the level and ignoring the `useOverride` sounds like an issue.

IMO, you should instead encode the tuple into an int in the same way `ScalableSettingLevelParameter.levelAndOverride.set` does. (like creating a static function for this).

**Comment by TomasKiniulis** [comment]
Quality settings are switched now without hovering mouse on Scene/Game view, but views show black/gray with exception "IndexOutOfRangeException: Index was outside the bounds of the array." when Custom quality setting is set
custom

**Pull request merged**